src/core/inverted_index_reader.rs

```diff
@@ -230,4 +230,13 @@ impl InvertedIndexReader {
         }
         Ok(())
     }
+
+    /// Returns the number of documents containing the term asynchronously.
+    pub async fn doc_freq_async(&self, term: &Term) -> crate::AsyncIoResult<u32> {
```

Comment:
In the future, rust may support to be generic over async. That would make things much easier for us

https://blog.rust-lang.org/inside-rust/2022/07/27/keyword-generics.html

Reply:
Yes indeed this is needed here :+1: